uat/mqtt-client-control/src/main/java/com/aws/greengrass/testing/mqtt/client/control/implementation/addon/MqttMessageEvent.java

```diff
@@ -118,6 +125,17 @@ private boolean isRetainMatched(Boolean retain) {
         return retain == null || retain == message.getRetain();
     }
 
+    private boolean isUserPropertiesMatched(List<Mqtt5Properties> userProperties) {
+        if (userProperties == null) {
```

Comment:
If argument userProperties (from filter) is null that means - we does not want to check user properties in received message.

Please add unit tests to check it.

Unit tests for retain is already added, another example is here:
https://github.com/Klika-Tech/aws-greengrass-client-device-auth/pull/84/files#diff-adbf31335af75d632627d69f28d8d169ce063414224b5916799a422f6f9c7e0cR359